### ISSUE TITLE
Redis Sentinel Config Unclear

### ISSUE THREAD

**Issue by kbrs-miaha** [TykTechnologies/tyk-helm-chart#217]
The impression from the `values.yaml` description regarding *Redis Sentinel* use, is to provide three config options, but does not make clear that the address also needs to be updated to reflect the *Sentinel* port (typically `26379`).  This is the relevant extract:

``` yaml
redis:
  # The addrs value will allow you to set your Redis addresses. If you are
  # using a redis cluster, you can list the endpoints of the redis instances
  # or use the cluster configuration endpoint.
  # Default value: redis.{{ .Release.Namespace }}.svc.cluster.local:6379
  # addrs:
  #   - redis.tyk.svc.cluster.local:6379
  #   This is the the DNS name of the redis as set when using Bitnami
  #   - tyk-redis-master.tyk.svc.cluster.local:6379

  # Redis password
  # If you're using Bitnami Redis chart please input your password in the field below
  # pass: ""

  # Enables SSL for Redis connection. Redis instance will have to support that.
  # Default value: false
  # useSSL: true

  # The enableCluster value will allow you to indicate to Tyk whether you are
  # running a Redis cluster or not.
  # enableCluster: true

  # Enables sentinel connection mode for Redis. If enabled, sentinelPass and masterName values are required.
  # enableSentinel: false

  # Redis sentinel password, only required while enableSentinel is true.
  # sentinelPass: ""

  # Redis sentinel master name, only required while enableSentinel is true.
  # masterName: ""
```

For Redis sentinel:
* Enable it: `enableSentinel: true`
* State password: `sentinelPass: "<password>"`
* Specify `masterName`: "redis-master"

The missing component, is that the address needs to include the Sentinel port:

```yaml
addrs:
  - tyk-redis-master.tyk.svc.cluster.local:26379
```

I found the reference to Bitnami not as helpful; it would be better to provide a more full example or be more specific about which bitnami chart you are referring to e.g. [Bitnami Redis Github](https://github.com/bitnami/charts/tree/master/bitnami/redis/).

In terms of potential solutions, the easiest might be change the following block of comments:

```
 # The addrs value will allow you to set your Redis addresses. If you are
  # using a redis cluster, you can list the endpoints of the redis instances
  # or use the cluster configuration endpoint.
# If using Redis Sentinel, update address to include Sentinel port (typical 26379)
```

And/or add another example at the bottom of the block:

```
  # addrs:
  #   - redis.tyk.svc.cluster.local:6379
  #   This is the the DNS name of the redis as set when using Bitnami
  #   - tyk-redis-master.tyk.svc.cluster.local:6379
 #   Sentinel port example
   #   - tyk-redis-master.tyk.svc.cluster.local:26379
```
Thanks!

Ashraf

**Comment by buraksekili** [comment]
Hi @kbrs-miaha,

Thank you for raising this issue! Your suggestions look great and will be helpful for other users as well. Would you like to open a PR, including your potential solution?

**Comment by kbrs-miaha** [comment]
@buraksekili - I have raised a PR, however on closer examination I noticed an inconsistency between the documentation (https://tyk.io/docs/tyk-oss/ce-helm-chart/) and the use of the chart.  The docs assume a *Redis Cluster* but cite the use of `bitnami/redis` chart, which supports *Redis* and *Redis Sentinel* only.  *Redis Cluster* is supported by the `bitnami/redis-custer` chart only.

In my proposal I make my one breaking change to align with the `bitnami/redis` chart, which is to set the `enableCluster` value to `false`.  The alternative is to leave this as `true` but then update all the documentation to refer to `bitnami/redis-cluster`.
